pkg/virt-launcher/notify-client/client.go

```diff
@@ -151,19 +155,32 @@ func libvirtEventCallback(c cli.Connection, domain *api.Domain, libvirtEvent lib
 	}
 }
 
-func (c *NotifyClient) StartDomainNotifier(domainConn cli.Connection, deleteNotificationSent chan watch.Event, vmiUID types.UID) error {
-
+func (c *NotifyClient) StartDomainNotifier(domainConn cli.Connection, deleteNotificationSent chan watch.Event, vmiUID types.UID, qemuAgentPollerInterval *time.Duration) error {
 	eventChan := make(chan libvirtEvent, 10)
 	reconnectChan := make(chan bool, 10)
 
 	domainConn.SetReconnectChan(reconnectChan)
 
+	qemuAgentCallback := func(domainName string, interfaceStatuses *[]api.InterfaceStatus) {
+		eventCallback(domainConn, util.NewDomainFromName(domainName, vmiUID), libvirtEvent{}, c, deleteNotificationSent, interfaceStatuses)
+	}
+	agentPoller := agentpoller.CreatePoller(domainConn, vmiUID, qemuAgentCallback, qemuAgentPollerInterval)
+
 	// Run the event process logic in a separate go-routine to not block libvirt
 	go func() {
 		for {
 			select {
 			case event := <-eventChan:
```

Comment:
Here we can have another case, where we listen for poll changes. If data changed, set a variable here and call the callback. Something like this:

```go
inferfaces = nil
for {
  select {
    case event := <-eventChan:
      # current code
      eventCallback(domainConn, domain, event, c, deleteNotificationSent, nil)
    case agentUpdate := <- agentStateChan:
      interfaces = agentUpdate.Interfaces
      eventCallback(domainConn, domain, event, c, deleteNotificationSent, nil)      
  }
}

```

Reply:
I think it is preferable to not distribute callback invocations across the code for sending virt-handler updates.